apollo-router/src/plugins/authentication/subgraph_auth.rs

```diff
@@ -0,0 +1,455 @@
+use std::collections::HashMap;
+use std::ops::ControlFlow;
+use std::sync::Arc;
+use std::time::SystemTime;
+
+use aws_credential_types::provider::ProvideCredentials;
+use aws_credential_types::Credentials;
+use aws_sigv4::http_request;
+use aws_sigv4::http_request::sign;
+use aws_sigv4::http_request::PayloadChecksumKind;
+use aws_sigv4::http_request::SignableBody;
+use aws_sigv4::http_request::SignableRequest;
+use aws_sigv4::http_request::SigningSettings;
+use aws_types::region::Region;
+use schemars::JsonSchema;
+use serde::Deserialize;
+use tower::BoxError;
+use tower::ServiceBuilder;
+use tower::ServiceExt;
+
+use crate::layers::ServiceBuilderExt;
+use crate::plugin::Plugin;
+use crate::plugin::PluginInit;
+use crate::register_plugin;
+use crate::services::subgraph;
+use crate::services::SubgraphRequest;
+
+register_plugin!("apollo", "subgraph_authentication", SubgraphAuth);
```

Comment:
Let's try and think holistically about the config:

We already have an authentication section in the config. Suggest:

```yaml
authentication:
  router:
    jwt: stuff
  subgraph:
    all:
      jwks: stuff
```

Reply:
I've put the changes in a separate branch https://github.com/apollographql/router/compare/igni/subgraph_auth...igni/move_auth_configuration?expand=1, I'm not satisfied with the diff, happy to discuss tomorrow.

Reply:
fixed in [dc15f32](https://github.com/apollographql/router/pull/3365/commits/dc15f32135deb3e421ea06fbbcf62e74a85b28ab)